### PR TITLE
feat(Button): Make initial component

### PR DESCRIPTION
**Pull request by michaeltaranto**
Initial pass at implementing the `Button` component. After taking both the [`seek-asia-style-guide`](https://seekinternational.github.io/seek-asia-style-guide/button/) and [`seek-style-guide`](https://seek-oss.github.io/seek-style-guide/button/) as inspiration and looking at how buttons are used across the three brands we have distilled our first pass down to three weights: `regular`, `weak` and `strong`.

Example:
```
<Button type="weak|regular|strong">Submit</Button>
```
By default `<Button>Submit</Button>` will render a `regular` button. See the docs site and playroom for more examples:
- [Playroom example](https://braid-design-system--f2338f2ffdafb60a6276ef5c26651caa6c6aceca.surge.sh/playroom/#?code=PGRpdiBzdHlsZT17eyBiYWNrZ3JvdW5kQ29sb3I6ICIjZjJmMmYyIiwgbWluSGVpZ2h0OiAiMTAwdmgiIH19PgogIDxDb2x1bW5zPgogICAgPENvbHVtbj4KICAgICAgPENhcmQ-CiAgICAgICAgPEhlYWRpbmcgbGV2ZWw9IjEiPkNhcmQgMTwvSGVhZGluZz4KICAgICAgICA8Qm94IHBhZGRpbmdUb3A9InNtYWxsIiBwYWRkaW5nQm90dG9tPSJsYXJnZSI-CiAgICAgICAgICA8VGV4dD5EbyB5b3UgbGlrZSBpdD88L1RleHQ-CiAgICAgICAgPC9Cb3g-CiAgICAgICAgPFJhZGlvIGNoZWNrZWQ9e3RydWV9IGxhYmVsPSJZZXMiIG1lc3NhZ2U9e2ZhbHNlfSAvPgogICAgICAgIDxSYWRpbyBsYWJlbD0iTm8iIC8-CiAgICAgICAgPEJ1dHRvbj5SZWd1bGFyIEJ1dHRvbjwvQnV0dG9uPgogICAgICA8L0NhcmQ-CiAgICA8L0NvbHVtbj4KICAgIDxDb2x1bW4-CiAgICAgIDxDYXJkPgogICAgICAgIDxIZWFkaW5nIGxldmVsPSIxIj5DYXJkIDI8L0hlYWRpbmc-CiAgICAgICAgPEJveCBwYWRkaW5nVG9wPSJzbWFsbCIgcGFkZGluZ0JvdHRvbT0ibGFyZ2UiPgogICAgICAgICAgPFRleHQ-SGVyZSBpcyBhbm90aGVyIGZvcm08L1RleHQ-CiAgICAgICAgPC9Cb3g-CiAgICAgICAgPENoZWNrYm94IGNoZWNrZWQ9e3RydWV9IGxhYmVsPSJTZWxlY3RlZCBvcHRpb24iIC8-CiAgICAgICAgPEJ1dHRvbiB3ZWlnaHQ9InN0cm9uZyI-U3Ryb25nIEJ1dHRvbjwvQnV0dG9uPgogICAgICA8L0NhcmQ-CiAgICA8L0NvbHVtbj4KICAgIDxDb2x1bW4-CiAgICAgIDxDYXJkPgogICAgICAgIDxIZWFkaW5nIGxldmVsPSIxIj5DYXJkIDM8L0hlYWRpbmc-CiAgICAgICAgPEJveCBwYWRkaW5nVG9wPSJzbWFsbCIgcGFkZGluZ0JvdHRvbT0ieGxhcmdlIj4KICAgICAgICA8VGV4dD5Mb3JlbSBpcHN1bSBkb2xvciBzaXQgYW1ldCwgY29uc2VjdGV0dXIgYWRpcGlzY2luZyBlbGl0LiBWZXN0aWJ1bHVtIG5vbiBlcm9zIGEgcXVhbSBtYXR0aXMgZmF1Y2lidXMgYXQgdWx0cmljaWVzIGZlbGlzLiBBbGlxdWFtIHNlbXBlciwgcXVhbSBldSBpbnRlcmR1bSBsdWN0dXMsIG1pIGxlY3R1cyBpbnRlcmR1bSBzYXBpZW4sIG5lYyBmcmluZ2lsbGEgdmVsaXQgaXBzdW0gdXQgZXguIEFlbmVhbiBlbmltIHVybmEsIGVsZW1lbnR1bSBncmF2aWRhIGxhY3VzIG5lYywgZmluaWJ1cyBwcmV0aXVtIHVybmEuIDwvVGV4dD4KICAgICAgICA8L0JveD4KICAgICAgICA8QnV0dG9uIHdlaWdodD0id2VhayI-V2VhayBCdXR0b248L0J1dHRvbj4KICAgICAgPC9DYXJkPgogICAgPC9Db2x1bW4-CiAgPC9Db2x1bW5zPgo8L2Rpdj4K)
- [Braid docs site](https://braid-design-system--f2338f2ffdafb60a6276ef5c26651caa6c6aceca.surge.sh/components/Button)

A few experiments in here that seem to be holding up but may revisit/tweak:
- ~Using box shadow in place of borders to prevent having to alter elements padding/size to get back onto grid/vertical rhythm.~ (Extracted and merged separately here https://github.com/seek-oss/braid-design-system/pull/131)
- Powering button styles from only `brandAccent` and `formAccent` colour and calculating variants statically for hover and active states.
- Using an overlay mask technique for affordance on weak buttons.

### CODE REVIEW

**Comment by mengtzu** [comment]
`Using box shadow in place of borders to prevent having to alter elements padding/size to get back onto grid/vertical rhythm.` 👏

**Comment by markdalgleish** [comment]
@mengtzu We've had some initial chats, and seems very positive so far—enough to allow this work into master, given its alpha state.

**Comment by patrickliu-sa** [comment]
Or can we use `outline` instead of `border` for buttons (while `white-space: nowrap` is set)?

**Comment by markdalgleish** [comment]
@patrickliu-sa we use an inset shadow, rather than a border/outline, so that the button stays on grid.

**Comment by michaeltaranto** [comment]
FYI I have force pushed over the top of this branch. After pulling out a lot of the other concerns that were originally in this PR, type clean ups, switching border to box shadows etc, i have cleaned this up to solely be about introducing `Button`.

File: lib/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    weak: 'formAccentActive',
+    regular: 'formAccentActive',
+    strong: 'brandAccentActive',
+  },
+};
+
+const foregroundColor: Record<ButtonWeight, Color> = {
+  weak: 'formAccent',
+  regular: 'white',
+  strong: 'brandAccentForeground',
```
Comment (markdalgleish) [inline lib/components/Button/Button.tsx]: Looking back at this with fresh eyes—I really love this 👏

**Pull request merged**